polaris/benchmark/_base.py

```diff
@@ -310,6 +294,47 @@ def _compute_checksum(dataset, target_cols, input_cols, split, metrics):
         checksum = hash_fn.hexdigest()
         return checksum
 
+    def verify_checksum(self, md5sum: Optional[str] = None):
+        """
+        Recomputes the checksum and verifies whether it matches the stored checksum.
+        """
+        if md5sum is None:
+            md5sum = self._md5sum
+        if md5sum is None:
+            logger.warning(
+                "No checksum to verify against. Specify either the md5sum parameter or "
+                "store the checksum in the benchmark.md5sum attribute. Skipping!"
+            )
+            return
+
+        # Temporarily reset
+        # Calling self.md5sum will recompute the checksum and set it again
```

Comment:
It seems like this might the kind of thing that becomes confusing in a few months once this isn't so fresh anymore, it's a little hard on the head to follow what `md5sum` is set to and what's being computed or set -- could we use `self._compute_checksum()` here explicitly to make it obvious what's going on?

Reply:
I think this could also potentially leave the benchmark with no `md5sum` set? Is that intentional? I.e. in the case where the `_md5sum` gets nulled out but the check fails, now the old `_md5sum` value will be lost and the benchmark's `_md5sum` will remain `None`, so in the case where the check fails `verify_checksum` is effectively resetting the `_md5sum` to `None`, which seems like it might be a bit surprising.